src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1512,9 +1508,12 @@ private bool HeadImprovementRequirementsSatisfied(BlockHeader header)
 
         private bool BestSuggestedImprovementRequirementsSatisfied(BlockHeader header)
         {
+            if (BestSuggestedHeader == null) return true;
+
             bool reachedTtd = header.IsPostTTD(_specProvider);
             bool isPostMerge = header.IsPoS();
-            bool tdImproved = header.TotalDifficulty > (BestSuggestedBody?.TotalDifficulty ?? 0);
+            bool tdImproved = (header.TotalDifficulty == (BestSuggestedBody?.TotalDifficulty ?? 0) && header.Number >= BestSuggestedBody?.Number)
```

Comment:
We're changing PoW behaviour in this place. Is it needed?

Reply:
Not sure, checking...

Reply:
Does not seems to be needed.